### ISSUE TITLE
non all-day multi-day events in week view only displayed as bar

### ISSUE THREAD

**Issue by Trikas77** [SimpleMobileTools/Simple-Calendar#506]
Example event:
Start 16:00 24 May '18
End 00:00 25 May '18 (or anything greater)

Displayed in Weekview:
Only displayed in top bar (all day events)

Expected:
If time not covering whole day: display visible times in week view as regular events + top bar for days where whole day is covers (or as whole day vertical bar)

**Comment by tibbi** [comment]
that is currently the expected behaviour, I dont want to fill the whole calendar with one event

**Comment by Trikas77** [comment]
When I have an event from 22:00 24 May to 06:00 25 May,
Wouldn't I expect to see an event on 24 May from 22:00 - 23:59 and an event on May 25 from 00:00 - 06:00?

**Comment by tibbi** [comment]
Currently it works as desired, I will see in the future if it needs to be changed. On the weekly view you would barely see an event from 22:00 to 6:00 by default at all as it shows days only from 7:00 to 23:00.

**Comment by robinpaulson** [comment]
The suggestion made by @Trikas77 would suit me too. I have a lot of events (maybe two per week) which go across midnight but are only 4 hours long. Currently they display as a bar which is imprecise and for me defeats the point of the grid format.

**Comment by robinpaulson** [comment]
@tibbi have you been able to take a further look at this? Thanks

**Comment by tibbi** [comment]
not really

**Comment by robinpaulson** [comment]
> not really

by "not really", do you mean "not at all"?

it'd be better if we know you're not going to fix this, it's an irritating bug, enough to make me switch to another calendar software.

on the other hand, if you're going to implement if any time soon....

**Comment by tibbi** [comment]
it likely wont be changed anytime soon, people seem to like this behaviour. I dont remember others reporting it.

**Comment by tibbi** [comment]
what app version are you using?

**Comment by dumblob** [comment]
@tiziodcaio you might be interested in this recent PR: https://github.com/SimpleMobileTools/Simple-Calendar/pull/1889

**Comment by qxzcode** [comment]
This is an issue for me too. I'm on the latest version, 6.20.3 from F-Droid.

Here's an example in my calendar. There is a period of 3 days that I have marked with an "All-day" event. This should display at the top, because it's obnoxious if it fills the full 72 hours in the grid. Conceptually, I'm using it to mark the *days/dates*, not specific time ranges within those days, so showing it in the grid is not helpful and visually conflicts with other events I have scheduled within those days.

I also have an event for a red-eye flight that spans midnight (say from 23:00 to 04:00 the next day). This should display in the grid, not at the top, so that I can see its specific timing relative to my other events in the day.

Basically, the behavior I personally desire is for "All-day" events to always display at the top (they are conceptually marking ranges of dates) and non-All-day events to always display in the grid (they are conceptually marking ranges of hours/minutes). If I remember correctly, this used to be the app's behavior until some recent update? I also like @thiswillbeyourgithub's suggestions.

**Comment by andy-berg** [comment]
I also think the old app behavior (< 6.20.2) regarding the display of multi-day all-day events should be restored, also because many stock calendars do it this way. The behavior of release 6.20.3 is a major issue for me for exactly the same reasons as described by @qxzcode
